lib/suse-cloud-upgrade-4-to-5-pre

```diff
@@ -206,7 +206,31 @@ for node in $allocated_nodes; do
 done
 
 # Make sure all openstack services are down on all nodes before we stop rabbitmq
-sleep 10
+services_stopped=""
+count=0
+while [ "x$services_stopped" != "xyes" -a "$count" -lt 100 ]; do
+  echo "Waiting for OpenStack services to stop..."
+  sleep 1
+  services_stopped="yes"
+  for node in $allocated_nodes; do
+    if ssh "$node" 'shopt -s nullglob
+      for i in /etc/init.d/openstack-* ; do
+        if $i status >/dev/null ; then
+          echo "$i still running at `hostname`"; exit 0
+        fi
+      done
+      exit 1'; then
+      services_stopped="no"
+    fi
+    if [ "x$services_stopped" == "xno" ]; then
```

Comment:
So to be clear:

    if [ $services_stopped == no ]; then

will work fine, although stylistically I personally prefer to quote the `'no'` because then your editor will syntax-highlight it as a string literal.